src/core/MOM_density_integrals.F90

```diff
@@ -824,226 +896,275 @@ subroutine int_density_dz_generic_ppm(k, tv, T_t, T_b, S_t, S_b, e, &
   use_covarTS = .false.
   use_varS = .false.
   if (use_stanley_eos) then
-     use_varT = associated(tv%varT)
-     use_covarTS = associated(tv%covarTS)
-     use_varS = associated(tv%varS)
+    use_varT = associated(tv%varT)
+    use_covarTS = associated(tv%covarTS)
+    use_varS = associated(tv%varS)
   endif
 
   T25(:) = 0.
   TS5(:) = 0.
   S25(:) = 0.
+  T215(:) = 0.
+  TS15(:) = 0.
+  S215(:) = 0.
 
   do n = 1, 5
     wt_t(n) = 0.25 * real(5-n)
     wt_b(n) = 1.0 - wt_t(n)
   enddo
 
+  ! Set the loop ranges for equation of state calculations at various points.
+  EOSdom_h5(1) = 1 ; EOSdom_h5(2) = 5*(Ieq-Isq+2)
+  EOSdom_q15(1) = 1 ; EOSdom_q15(2) = 15*(Ieq-Isq+1)
+  EOSdom_h15(1) = 1 ; EOSdom_h15(2) = 15*(HI%iec-HI%isc+1)
+
   ! 1. Compute vertical integrals
-  do j=Jsq,Jeq+1 ; do i=Isq,Ieq+1
-    if (use_PPM) then
-      ! Curvature coefficient of the parabolas
-      s6 = 3.0 * ( 2.0*tv%S(i,j,k) - ( S_t(i,j,k) + S_b(i,j,k) ) )
-      t6 = 3.0 * ( 2.0*tv%T(i,j,k) - ( T_t(i,j,k) + T_b(i,j,k) ) )
-    endif
-    dz = e(i,j,K) - e(i,j,K+1)
-    do n=1,5
-      p5(n) = -GxRho*((e(i,j,K) - z0pres) - 0.25*real(n-1)*dz)
-      ! Salinity and temperature points are reconstructed with PPM
-      S5(n) = wt_t(n) * S_t(i,j,k) + wt_b(n) * ( S_b(i,j,k) + s6 * wt_t(n) )
-      T5(n) = wt_t(n) * T_t(i,j,k) + wt_b(n) * ( T_b(i,j,k) + t6 * wt_t(n) )
+  do j=Jsq,Jeq+1
+    do i=Isq,Ieq+1
+      if (use_PPM) then
+        ! Curvature coefficient of the parabolas
+        s6 = 3.0 * ( 2.0*tv%S(i,j,k) - ( S_t(i,j,k) + S_b(i,j,k) ) )
+        t6 = 3.0 * ( 2.0*tv%T(i,j,k) - ( T_t(i,j,k) + T_b(i,j,k) ) )
+      endif
+      dz = e(i,j,K) - e(i,j,K+1)
+      do n=1,5
+        p5(I*5+n) = -GxRho*((e(i,j,K) - z0pres) - 0.25*real(n-1)*dz)
+        ! Salinity and temperature points are reconstructed with PPM
+        S5(I*5+n) = wt_t(n) * S_t(i,j,k) + wt_b(n) * ( S_b(i,j,k) + s6 * wt_t(n) )
+        T5(I*5+n) = wt_t(n) * T_t(i,j,k) + wt_b(n) * ( T_b(i,j,k) + t6 * wt_t(n) )
+      enddo
+      if (use_stanley_eos) then
+        if (use_varT) T25(I*5+1:I*5+n) = tv%varT(i,j,k)
```

Comment:
I believe these `I*5+1:I*5+n` loops should be `I*5+1:I*5+5`, but I haven't run any Stanley EOS tests.